node/network/gossip-support/src/lib.rs

```diff
@@ -64,6 +74,13 @@ struct State {
 	// at least a third of authorities the last time.
 	// `None` otherwise.
 	last_failure: Option<Instant>,
+
+	/// First time we did not reach our connectivity treshold.
+	///
+	/// This is the time of the first failed attempt to connect to >2/3 of all validators in a
+	/// potential sequence of failed attempts. It will be cleared once we reached >2/3
+	/// connectivity.
+	failure_start: Option<Instant>,
```

Comment:
Might worth pointing out that it is address resolution (`AuthorityDiscoveryId` -> `PeerId`) that has failed, in which case connectivity also suffers, but we don't know about if connectivity is good if it hasn't. 

I guess it is an Option to assume that the failure might happen at a random point in the future (e.g. at session boundaries). But that would be really unexpected unless everyone rotates their keys at the same time.